File: shared/schemas/word.ts
import { type Static, Type } from '@sinclair/typebox';

export const WordSchema = Type.Object({
  id: Type.String({ format: 'uuid' }),
  symbols: Type.String(),
  transcription: Type.String(),
  translation: Type.String(),
});

export const CreateWordSchema = Type.Omit(WordSchema, ['id']);

export type WordDto = Static<typeof WordSchema>;
export type CreateWordDto = Static<typeof CreateWordSchema>;

File: backend/internal/dto/word.go
package dto

type Word struct {
	ID            string `json:"id"`
	Symbols       string `json:"symbols"`
	Transcription string `json:"transcription"`
	Translation   string `json:"translation"`
}

type CreateWord struct {
	Symbols       string `json:"symbols"`
	Transcription string `json:"transcription"`
	Translation   string `json:"translation"`
}
